clippy_lints/src/indexing_slicing.rs

```diff
@@ -126,6 +126,10 @@ impl<'tcx> LateLintPass<'tcx> for IndexingSlicing {
                         return;
                     };
 
+                    if is_from_proc_macro(cx, expr) {
```

Comment:
Can we have this check earlier in `check_expr`?

Reply:
Yes, for sure! I thought that it would be better to postpone to the lastest moment, because this check is "heavy", but probably I misunderstood it :)

I will do the change!

Reply:
@dswij done. I have moved it to the very beginning of the function. Because of that, now we can have just one check instead of 4. Commit was amended.